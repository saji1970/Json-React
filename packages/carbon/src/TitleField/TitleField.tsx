import { FormContextType, RJSFSchema, StrictRJSFSchema, TitleFieldProps } from '@rjsf/utils';
import { LabelValue } from '../components/LabelValue';
import { FormGroup } from '@carbon/react';

/** Implement `TitleFieldTemplate`
 */
export default function TitleField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
  id,
  title,
  required,
}: TitleFieldProps<T, S, F>) {
  return (
    <div className='title-field'>
      <FormGroup legendId={id} legendText={<LabelValue label={title} required={required} hide={false} />}>
        {null}
      </FormGroup>
    </div>
  );
}
