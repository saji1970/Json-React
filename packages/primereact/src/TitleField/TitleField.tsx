import React from "react";
import { FieldProps } from "@rjsf/core";

const TitleField = ({ title, uiSchema }: Pick<FieldProps, "title" | "uiSchema">) => (
  <div className="border-bottom-1 mb-2">
    <h5>{(uiSchema && uiSchema["ui:title"]) || title}</h5>
  </div>
);

export default TitleField;
