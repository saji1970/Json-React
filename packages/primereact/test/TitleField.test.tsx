import React from "react";
import TitleField from "../src/TitleField";
import renderer from "react-test-renderer";

describe("TitleField", () => {
  test("simple", () => {
    const tree = renderer.create(<TitleField title="Hello" uiSchema={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
