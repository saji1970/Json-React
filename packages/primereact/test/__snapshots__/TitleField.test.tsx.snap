// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleField simple 1`] = `
<div
  className="border-bottom-1 mb-2"
>
  <h5>
    Hello
  </h5>
</div>
`;
